src/rpdk/guard_rail/core/stateful.py

```diff
@@ -89,16 +91,59 @@ class DIFFKEYS:
 }
 
 
+def remove_key(document: Dict[str, Any], key: str):
+    """removes only top level property"""
+    cloned_document = deepcopy(document)
+    if key in cloned_document:
+        del cloned_document[key]
+    return cloned_document
+
+
+def get_value(document: Dict[str, Any], key: str):
+    """retrieves only top level property"""
+    return document.get(key)
+
+
 def schema_diff(previous_json: Dict[str, Any], current_json: Dict[str, Any]):
     """schema diff function to get formatted schema diff from deep diff"""
-    deep_diff = DeepDiff(
-        resolve_schema(previous_json),
-        resolve_schema(current_json),
+
+    previous_schema = resolve_schema(previous_json)
+    current_schema = resolve_schema(current_json)
+
+    previous_schema_no_primary_id = remove_key(previous_schema, "primaryIdentifier")
+    current_schema_no_primary_id = remove_key(current_schema, "primaryIdentifier")
+
+    previous_schema_only_primary_id = {
+        "primaryIdentifier": get_value(previous_schema, "primaryIdentifier")
+    }
+    current_schema_only_primary_id = {
+        "primaryIdentifier": get_value(current_schema, "primaryIdentifier")
+    }
+
+    deep_diff_no_primary_id = DeepDiff(
```

Comment:
probably this splitting into 2 schemas is not required if `ignore_order_func` can be used to exclude only `primaryIdentifier` key 

```
def ignore_order_func(level):
    return 'primaryIdentifier' not in level.path()
...
DeepDiff(json1, json2,ignore_order=True, ignore_order_func=ignore_order_func)
```

Reply:
Ref:- https://zepworks.com/deepdiff/current/ignore_order.html#ignore-order-func-label